test/puppetlabs/stockpile/queue_test.clj

```diff
@@ -101,18 +102,32 @@
   (call-with-temp-dir-path
    (fn [tmpdir]
      (let [qdir (.toFile (.resolve tmpdir "queue"))
-           newq (stock/create qdir)]
-       (let [entry-1 (store-str newq "foo" "meta foo")
-             q (stock/open qdir)
-             read-entries (stock/reduce q conj ())]
-         (is (= [entry-1] read-entries))
-         (is (= "foo" (slurp-entry q entry-1)))
-         (let [entry-2 (store-str q "bar" "meta bar")]
-           (let [q (stock/open qdir)
-                 read-entries (stock/reduce q conj #{})]
-             (is (= #{entry-2 entry-1} read-entries))
-             (is (= "foo" (slurp-entry q entry-1)))
-             (is (= "bar" (slurp-entry q entry-2))))))))))
+           ent-1-id (let [q (stock/create qdir)
```

Comment:
Possibly, though this is also testing reduce with more than one item (with and without metadata) along the way; I can certainly disentangle things and see if the (expanded) result's preferable.